icons/icons.json

```
@@ -1692,7 +1692,7 @@
 					"kdl"
 				],
 				"name": "Plain Text (KDL)",
-				"scope": "source.kdl"
+				"scope": "text.kdl"
```

If no issues are to be expected by, I'd suggest to keep `source.kdl` for compatibility reasons with older syntax versions. A File Icon still supports ST3, which current version of KDL does not.

```suggestion
				"scope": "source.kdl, text.kdl"
```

---

The old version of KDL also didn't support ST3, it used ST4 v2 of the syntax since the beginning, so not sure why it would be an issue there?

> If no issues are to be expected by

The syntax override is the issues expected that prompted this PR: as I understood from the similar Fish issue scope mismatch is the cause

---

The scenaio I have in mind is users still using an old syntax with `source.kdl` while upgrading A File Icon. Those would run into same kind of issues due to changed scopes. The proposed change would ensure both scopes are accepted and alias syntaxes not being created.

If KDL however supports all stable ST4107+ builds, that's probably more of a theoretical concern and we can go with `text.kdl` only.

---

I'm not aware of any build requirement outside of v2 syntax since ST 4075, same as the previous versions of KDL, so the new version should work the same